ceremonies/src/lib.rs

```diff
@@ -575,6 +582,32 @@ impl<T: Config> Module<T> {
 		})
 	}
 
+	fn purge_inactive_communities() {
+		let cindex = <encointer_scheduler::Module<T>>::current_ceremony_index();
+
+		for cid in <encointer_communities::Module<T>>::community_identifiers() {
+			if AssignmentCounts::get((cid, cindex)).get_number_of_participants() == 0 {
+				let current = InactivityCounters::get(cid);
+				if current > T::InactivityTimeout::get() {
+					Self::purge_community(cid);
+				} else {
+					InactivityCounters::insert(cid, current + 1);
+				}
+			} else {
+				InactivityCounters::insert(cid, 0);
+			}
+		}
+	}
+
+	fn purge_community(cid: CommunityIdentifier) {
+		for cindex in 1..<encointer_scheduler::Module<T>>::current_ceremony_index() {
```

Comment:
No need to start with 1 here. Too expensive. the registry is purged after reputation lifetime anyway